### DISCUSSION TITLE
Latex-tree view package not working

### DISCUSSION THREAD

**Discussion by stefaniecg** [pulsar-edit/.github#135] [Community Package Help]
The package [Latex-tree-view](https://web.pulsar-edit.dev/packages/latex-tree) is not working in Pulsar, as it used to work in Atom. 
It generates the list of sections, but the links are not clickable any more. 
Before in Atom, it used to jump to the clicked section but now in pulsar it doesn't.

**Comment by confused-Techie** [comment]
Does this package work in Atom `1.60.0`?

**Comment by stefaniecg** [comment]
Hi @confused-Techie, 

Yes, i just checked the version of my **Atom** and it is `1.60.0`

![Screenshot 2023-02-19 at 12 34 33](https://user-images.githubusercontent.com/64597312/219945402-104dbc74-15c9-491e-b1a6-d18781b5c7ba.png)

**Reply by confused-Techie** [reply]
Appreciate you double checking!

Also appreciate that you've already created an issue in their repo. As it's possible just by updating Node and Electron is what caused their package to break. In which case they would need to fix it. But if it is something on our end hopefully we can find out.

**Comment by savetheclocktower** [comment]
@stefaniecg I suspected it was a quick fix, so I took a look and found the problem. The good news is that it is a quick fix after all, and I opened [this PR](https://github.com/raphael-cch/latex-tree/pull/21) to fix it.

The bad news is that the maintainer hasn't been active in that repository since 2018. I would imagine there is a 0% chance that they will see that PR and put out a new version.

You have a couple options:

1. If you're at all savvy with JavaScript, you can view the changes I made in that PR and make the same changes locally. Your copy of `latex-tree` is stored within your Pulsar home folder, but I don't know what platform you're on, so the quickest way to figure out where that is is to open the dev tools (Ctrl-Shift-I on Windows/Linux, Cmd-Opt-I on macOS) and type `process.env.ATOM_HOME` in the console. That will give you a location on disk. From there, you can go into `packages/latex-tree/lib/latex-tree-view.js` and make the changes. After the changes are made, relaunch Pulsar and the package should be fixed.
2. If you're not confident enough to make that change, you can install my fork of `latex-tree` from the command line. But if you want to do that, I'll ask you to tell me which platform you're on, and we can proceed from there.

**Comment by savetheclocktower** [comment]
@stefaniecg OK, I see in your profile that you starred a QuickLook plugin repo, so I'll assume you're on macOS. :-)

1. Go to **Pulsar → Install Shell Commands** from the menu to ensure that `pulsar` and `ppm` are installed. 
2. Open your Pulsar settings, go to the Packages section, and uninstall `latex-tree`.
3. Open your terminal and run `ppm install savetheclocktower/latex-tree`.
4. Restart Pulsar.

I don't imagine I'll keep my fork of `latex-tree` around indefinitely, but once you install it from there, it won't matter.

Let me know if that fixes it.

**Reply by stefaniecg** [reply]
@Andrew, hi, thank you for your fix. 
I just tried to do run the `ppm install savetheclocktower/latex-tree`
but it gives me an error, so not sure what i'm doing wrong. 

Take a look at my terminal, and let me know what is the mistake from my side. 

Thank you again !

**Reply by savetheclocktower** [reply]
That's odd. Works when I try it. OK, let's do this instead:

* Go to [this page](https://github.com/savetheclocktower/latex-tree/releases/tag/v0.5.1a) and download the ZIP file.
* Extract the ZIP file and rename the folder from `latex-tree-0.5.1a` to `latex-tree`.
* Open another Finder window, hit Cmd-Shift-G, and type in `~/.pulsar/packages`.
* Drag the `latex-tree` folder over into the new finder window.
* Restart Pulsar.

If that doesn't work, delete the new `latex-tree` folder and try this:

* Reinstall your original `latex-tree` package by searching for it within Pulsar.
* Open `~/.pulsar/packages/latex-tree/lib/` in Finder.
* Re-extract `latex-tree-0.5.1a` and go into `latex-tree-0.5.1a/lib` to find the fixed `latex-tree-view.js` file.
* Drag it over to the other `lib` folder and tell Finder to replace the file.
* Restart Pulsar.

**Reply by stefaniecg** [reply]
@savetheclocktower (aka Andrew) Thank you so much for this fix. 
I executed the first set of steps and now the latex-tree works as it used to in Atom. 
This will help me so so much as i use latex so much. 
Thank you for your help.
